pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -503,24 +503,37 @@ func (c *Reconciler) updateLabels(tr *v1alpha1.TaskRun) (*v1alpha1.TaskRun, erro
 	return newTr, nil
 }
 
-// createPod creates a Pod based on the Task's configuration, with pvcName as a
-// volumeMount
-func (c *Reconciler) createPod(tr *v1alpha1.TaskRun, ts *v1alpha1.TaskSpec, taskName string) (*corev1.Pod, error) {
-	ts = ts.DeepCopy()
 
-	err := resources.AddOutputImageDigestExporter(tr, ts, c.resourceLister.PipelineResources(tr.Namespace).Get)
+// createPod creates a Pod based on the Task's configuration, with pvcName as a volumeMount
+// TODO(dibyom): Refactor resource setup/templating logic to its own function in the resources package
+func (c *Reconciler) createPod(tr *v1alpha1.TaskRun, rtr *resources.ResolvedTaskResources) (*corev1.Pod, error) {
+	ts := rtr.TaskSpec.DeepCopy()
+	inputResources, err := resourceImplBinding(rtr.Inputs)
+	if err != nil {
+		c.Logger.Errorf("Failed to initialize input resources: %v", err)
+		return nil, err
+	}
+	outputResources, err := resourceImplBinding(rtr.Outputs)
+	if err != nil {
+		c.Logger.Errorf("Failed to initialize output resources: %v", err)
+		return nil, err
```

Comment:
i think we usually lean toward a pattern where we wrap the error with more context and return it - and only in the reconciler loop where the return value of `createPod` is handled would we log

i'm happy to leave this as-is for now tho and we can improve it later :D (esp. if you are already planning to refactor :innocent: )